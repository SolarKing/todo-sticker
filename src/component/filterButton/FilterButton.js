// @flow
import React from 'react';
import { Button } from 'antd';
import Translate from '../../class/translate';
import FilterPanel from '../filterPanel/FilterPanel';

type Props = {

}

type States = {
    showPanel: boolean,
}

export default class FilterButton extends React.Component<Props, States> {

    state: States = {
        showPanel: false,
    }

    constructor(props: Props) {
        super(props);
    }

    render() {
        return (
            <div {...this.props}>
                <Button onClick={() => this.setState({showPanel: !this.state.showPanel})} >{Translate.tr('filter')}</Button>
                <FilterPanel show={this.state.showPanel} />
            </div>
        );
    }
}
