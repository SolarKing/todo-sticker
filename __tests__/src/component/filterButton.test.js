// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import FilterButton from '../../../src/component/filterButton';
import { Button } from 'antd';

describe('FilterButton', () => {
    let wrapper: ShallowWrapper;
    let onClick = jest.fn();
    beforeAll(() => {
        wrapper = shallow(<FilterButton onClick={onClick} />);
    });

    it('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should trigger onClick', () => {
        wrapper.find(Button).simulate('click');
        expect(onClick.mock.calls.length).toEqual(1);
    });

});
