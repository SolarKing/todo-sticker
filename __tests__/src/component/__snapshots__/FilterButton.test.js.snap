// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton should render correct 1`] = `
<Button
  ghost={false}
  loading={false}
  onClick={[MockFunction]}
  prefixCls="ant-btn"
>
  filter
</Button>
`;
